components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodManager.kt

```diff
@@ -199,6 +205,105 @@ class GitpodManager : Disposable {
+                            lifetime.onTerminationOrNow {
+                                requestStream.cancel(null, null)
+                            }
+                        }
+
+                        override fun onNext(ps: PortsStatusResponse) {
+                            for (port in ps.portsList) {
+                                // Avoiding undesired notifications
+                                if (ignorePorts.contains(port.localPort)) {
+                                    continue
+                                }
+
+                                val hasPreviousStatus = portsStatus.containsKey(port.localPort)
+
+                                if (!hasPreviousStatus) {
+                                    portsStatus[port.localPort] = port
+                                }
+
+                                val wasServed = portsStatus[port.localPort]?.served!!
+                                val wasExposed = portsStatus[port.localPort]?.hasExposed()!!
+                                val wasServedExposed = wasServed && wasExposed
+                                val isServedExposed = port.served && port.hasExposed()
+
+                                // If the initial update received shows that the port is served and exposed, then notify
+                                val isFirstUpdate = !hasPreviousStatus && wasServedExposed && isServedExposed
+
+                                // If the port changes its status to served and exposed, notify the user
+                                val shouldSendNotification = isFirstUpdate || !wasServedExposed && isServedExposed
+
+                                portsStatus[port.localPort] = port
+
+                                if (shouldSendNotification) {
+                                    if (port.exposed.onExposed.number == OnPortExposedAction.ignore_VALUE) {
+                                        continue
+                                    }
+
+                                    if (port.exposed.onExposed.number == OnPortExposedAction.open_browser_VALUE) {
+                                        BrowserUtil.browse(port.exposed.url)
+                                        continue
+                                    }
+
+                                    if (port.exposed.onExposed.number == OnPortExposedAction.open_preview_VALUE) {
+                                        BrowserUtil.browse(port.exposed.url)
+                                        continue
+                                    }
+
+                                    val message = "A service is available on port ${port.localPort}"
+                                    val notification = notificationGroup.createNotification(message, NotificationType.INFORMATION)
+                                    // TODO(andreafalzetti): add analytics event similar to https://github.com/gitpod-io/openvscode-server/blob/7c912399c70799e4e4452a63cf3c0f7cc8f5832c/extensions/gitpod-web/src/extension.ts#L582
```

Comment:
I couldn't find a setup for analytics in this component. Am I missing something or we don't have it yet?

Reply:
It is here https://github.com/gitpod-io/gitpod/blob/a60fc2d2a632651aab4ace5532515e4cc02f30ae/components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodClientProjectSessionTracker.kt#L62

Reply:
oooh! thanks @mustard-mh :) I've added analytics now 🚀

Reply:
We agreed with @akosyakov to remove the analytics event for now. It costs us $$$ and if we are not actively using it, it only causes noise.